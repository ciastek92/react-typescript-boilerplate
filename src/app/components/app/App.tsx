import Home from 'app/pages/Home';
import React from 'react';
import { Route } from 'react-router-dom';
import { ToastContainer } from 'react-toastify';
import Layout from '../layout/layout';

const App = () => (
  <Layout>
    <Route exact path="/" component={Home} />
    <ToastContainer />
  </Layout>
);

export { App };
