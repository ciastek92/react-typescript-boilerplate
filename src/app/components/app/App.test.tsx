import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import { App } from './App';

Enzyme.configure({
  adapter: new Adapter()
});

jest.mock(
  'react-router-dom',
  (): any => ({
    Route: (): string => 'Route'
  })
);

describe('App component tests', () => {
  Enzyme.configure({ adapter: new Adapter() });

  it('renders application correctly', () => {
    const tree = shallow(<App />);
    expect(tree).toMatchSnapshot();
  });
});
