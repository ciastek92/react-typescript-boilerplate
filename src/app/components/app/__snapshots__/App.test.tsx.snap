// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component tests renders application correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />,
        <ToastContainer
          autoClose={5000}
          bodyClassName={null}
          className={null}
          closeButton={
            <CloseButton
              ariaLabel="close"
            />
          }
          closeOnClick={true}
          draggable={true}
          draggablePercent={80}
          hideProgressBar={false}
          newestOnTop={false}
          pauseOnFocusLoss={true}
          pauseOnHover={true}
          position="top-right"
          progressClassName={null}
          progressStyle={null}
          rtl={false}
          style={null}
          toastClassName={null}
          transition={[Function]}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoClose": 5000,
          "bodyClassName": null,
          "className": null,
          "closeButton": <CloseButton
            ariaLabel="close"
          />,
          "closeOnClick": true,
          "draggable": true,
          "draggablePercent": 80,
          "hideProgressBar": false,
          "newestOnTop": false,
          "pauseOnFocusLoss": true,
          "pauseOnHover": true,
          "position": "top-right",
          "progressClassName": null,
          "progressStyle": null,
          "rtl": false,
          "style": null,
          "toastClassName": null,
          "transition": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />,
          <ToastContainer
            autoClose={5000}
            bodyClassName={null}
            className={null}
            closeButton={
              <CloseButton
                ariaLabel="close"
              />
            }
            closeOnClick={true}
            draggable={true}
            draggablePercent={80}
            hideProgressBar={false}
            newestOnTop={false}
            pauseOnFocusLoss={true}
            pauseOnHover={true}
            position="top-right"
            progressClassName={null}
            progressStyle={null}
            rtl={false}
            style={null}
            toastClassName={null}
            transition={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoClose": 5000,
            "bodyClassName": null,
            "className": null,
            "closeButton": <CloseButton
              ariaLabel="close"
            />,
            "closeOnClick": true,
            "draggable": true,
            "draggablePercent": 80,
            "hideProgressBar": false,
            "newestOnTop": false,
            "pauseOnFocusLoss": true,
            "pauseOnHover": true,
            "position": "top-right",
            "progressClassName": null,
            "progressStyle": null,
            "rtl": false,
            "style": null,
            "toastClassName": null,
            "transition": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
